### ISSUE TITLE
Separation module not compatible with NumPy 2.0

### ISSUE THREAD

**Issue by richa-namballa** [craffel/mir_eval#386]
`np.Inf` was replaced with `np.inf` in NumPy 2.0 so the [`_safe_db` ](https://github.com/craffel/mir_eval/blob/485a42551f449517567313f9871f58e9c99fbd8a/mir_eval/separation.py#L834) function breaks.

Here's the traceback of a dummy example.
```
>>> import numpy as np
>>> import mir_eval
>>> y_ref = np.random.randn(1, 48000)
>>> y_est = np.random.randn(1, 48000)
>>> mir_eval.separation.bss_eval_sources(y_ref, y_est)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/richanamballa/miniconda3/envs/mss/lib/python3.12/site-packages/mir_eval/separation.py", line 214, in bss_eval_sources
    _bss_source_crit(s_true, e_spat, e_interf, e_artif)
  File "/Users/richanamballa/miniconda3/envs/mss/lib/python3.12/site-packages/mir_eval/separation.py", line 810, in _bss_source_crit
    sir = _safe_db(np.sum(s_filt**2), np.sum(e_interf**2))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/richanamballa/miniconda3/envs/mss/lib/python3.12/site-packages/mir_eval/separation.py", line 833, in _safe_db
    return np.Inf
           ^^^^^^
  File "/Users/richanamballa/miniconda3/envs/mss/lib/python3.12/site-packages/numpy/__init__.py", line 397, in __getattr__
    raise AttributeError(
AttributeError: `np.Inf` was removed in the NumPy 2.0 release. Use `np.inf` instead.. Did you mean: 'inf'?
```

It should be an easy fix to replace `np.Inf` with `np.inf`.

For reference, I'm using `numpy 2.0.1` and `mir_eval 0.7`. As a temporary fix, I downgraded to `numpy 1.26.4`.

**Comment by richa-namballa** [comment]
Never mind! I see that the issue is being fixed in #385.

**Comment by carlthome** [comment]
Yeah, trying to get this fixed ASAP. ETA this week. 🤞
